faq.md

```diff
@@ -0,0 +1,86 @@
+
+## Test a configuration
+
+If a user wishes to verify that a configuration file is valid it is recommended to first pass the configuration through a YAML syntax validator. This is done to validate the syntactic validity of the file.
+
+Next, a user can run `sudo netplan generate` to validate the configuration is correctly parsed by netplan and written to disk. However, be warned that once generated the new configuration is written to disk the configuration is activated on next boot. Therefore, backing up a current working configuration is highly recommended.
+
+## Hierarchy of configuration files
+
+Configuration files can exist in three different locations with the precedence from most important to least as follows:
+
+* /run/netplan/*.yaml
+* /etc/netplan/*.yaml
+* /lib/netplan/*.yaml
+
+Alphabetically later files, no matter what directory in, will amend keys if the key does not already exist and override previous keys if they do.
+
+## Bring interfaces up or down
+
+Previously users were used to using the `ifconfig` command. Users should now familiarize themselves with the more powerful `ip` command. Manually modifying network devices is now accomplished via the ip command.
+
+As an example to bring up an interface and bring it back down:
+
+```shell
+ip link set enp3s0 up
+ip link set enp3s0 down
+```
+
+See `man ip` for more information on how to manipulate the state of routing, network devices, interfaces and tunnels.
+
+## Prevent waiting for interface
+
+Interfaces that are not required for booting or should not be waited for during boot and have the `optional: true` key added to them. This will prevent long delays in booting for interfaces that may not come up.
+
+## Get the current system address
+
+Netplan is only meant to translate configuration into the appropriate files for the backend specified. It does not display IP information by itself, since it does not manage those by itself.
+
+To get the most accurate state of the IP addresses for the system using the `ip addr` command.
+
+## Find the current DNS servers
+
+To determine the current DNS servers used by the system run `systemd-resolve --status` and look for the 'DNS Servers:' entry to see what DNS server is used.
+
+## Deconfigure an interface
+
+To deconfigure an interface, remove the configuration for the device from the netplan .yaml file and run `sudo netplan apply`.
+
+If the interface is not configured in a .yaml file in `/etc/netplan`, it will not be configured at boot. To remove addresses manually, a user can run `ip address del <address> dev <interface>`.
+
+## Use pre-up, post-up, etc. hook scripts
+
+Users of ifupdown may be familiar with using hook scripts (e.g pre-up, post-up, etc.) in their interfaces file. Netplan configuration does not include support for hook scripts. Instead to achieve this functionality users should add a systemd unit jobs with the appropriate 'Requires:' and 'After:' fields to run arbitrary commands once the network is up.
```

Comment:
See: https://askubuntu.com/questions/987982/how-can-i-run-custom-scripts-once-an-interface-comes-up/1014614#1014614

We should at least seed this with systemd-dispatcher language, can clean up with a better example later.